pystan/diagnostics.py

```diff
@@ -0,0 +1,436 @@
+        if e_bfmi[chain_num] < 0.2:
+            if verbosity > 0:
+                logger.warning('Chain {}: E-BFMI = {}'.format(chain_num + 1,
+                                                              e_bfmi[chain_num]))
+                
+            no_warning = False
+        else:
+            if verbosity > 1:
+                logger.info('Chain {}: E-BFMI (= {}) '.format(chain_num + 1,
+                                                              e_bfmi[chain_num]) +
+                            'equals or exceeds threshold of 0.2.')
+            
+    if no_warning:
+        if verbosity > 1:
+            logger.info('E-BFMI indicated no pathological behavior')
+            
+        return True
+    else:
+        if verbosity > 0:
+            logger.warning('E-BFMI below 0.2 indicates you may need to reparameterize your model')
+            
+        return False
+
+def check_n_eff(fit, verbose = True):
+    """Checks the effective sample size per iteration
+
+    Parameters
+    ----------
+    fit : StanFit4Model object
+    verbose : bool or int, optional
+        If ``verbose`` is ``False`` or a nonpositive integer, no
+        diagnostic messages are printed, and only the return value of
+        the function conveys diagnostic information. If it is ``True``
+        (the default) or an integer greater than zero, then a
+        diagnostic message is printed only if there are effective
+        sample sizes that appear pathologically low. If it is an
+        integer greater than 1, then extra diagnostic messages are
+        printed.
+
+
+    Returns
+    -------
+    bool
+        ``True`` if there are no problems with effective sample size
+        and ``False`` otherwise.
+
+    Raises
+    ------
+    ValueError
+        If the output of ``fit.summary()`` has no information about
+        effective sample size (i.e., n_eff).
+
+    """
+
+    verbosity = int(verbose)
+    
+    fit_summary = fit.summary(probs=[0.5])
+
+    try:
+        n_effs_index = fit_summary['summary_colnames'].index('n_eff')
+    except:
+        raise ValueError('Summary of fit object appears to lack information on effective sample size')
+        
+    n_effs = fit_summary['summary'][:, n_effs_index]
+    names = fit_summary['summary_rownames']
+    n_iter = len(fit.extract()['lp__'])
+
+    no_warning = True
+    for n_eff, name in zip(n_effs, names):
+        ratio = n_eff / n_iter
+        if ((ratio < 0.001) or np.isnan(ratio) or np.isinf(ratio)):
+            if verbosity > 0:
+                logger.warning('n_eff / iter for parameter {} is {}!'.format(name, ratio))
+                
+            no_warning = False
+            
+    if no_warning:
+        if verbosity > 1:
+            logger.info('n_eff / iter looks reasonable for all parameters')
+
+        return True
+    else:
+        if verbosity > 0:
+            logger.warning('n_eff / iter below 0.001 indicates that the effective sample size has likely been overestimated')
```

Comment:
Is return statement missing?
 
    return False

Reply:
Yes, it is missing. Good catch.